### DISCUSSION TITLE
SpeechBrain VAD implementation

### DISCUSSION THREAD

**Discussion by pranavtushar** [speechbrain/speechbrain#1911] [Q&A] [answered]
While using boundaries = VAD.get_speech_segments(path), I am getting an error "he detected sample rate is different from that set in the hparam file".

**Comment by asumagic** [comment, marked as answer]
`get_speech_segments` calls `get_speech_prob_file`, which includes that check. I am not sure why it is done like this, but it may be that the chunked reading would be sensitive to resampling. Ideally, if you have the possibility, you should resample your source files to the sample rate that is listed in the VAD hyperparameter file. If you use `speechbrain/vad-crdnn-libriparty`, this is 16kHz.

**Comment by pranavtushar** [comment]
Hey @AsuMagic , Thanks!
Yes resampling the audio to 16kHz and converting it to single channel audio and then passing the audio to ### get_speech_segments helps solve the error.
